tests/integration/test_log_rotation.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+from constants import CHARMED_MYSQL_COMMON_DIRECTORY
+
+from .helpers import (
+    app_name,
+    delete_file_or_directory_in_unit,
+    ls_la_in_unit,
+    read_contents_from_file_in_unit,
+    write_content_to_file_in_unit,
+)
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+TIMEOUT = 15 * 60
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest, mysql_charm_series: str) -> None:
+    """Build and test a unit of the charm."""
+    charm = await ops_test.build_charm(".")
+
+    await ops_test.model.deploy(
+        charm,
+        application_name=APP_NAME,
+        num_units=1,
+        series=mysql_charm_series,
+    )
+
+    # Reduce the update_status frequency until the cluster is deployed
+    async with ops_test.fast_forward("60s"):
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME],
+            status="active",
+            raise_on_blocked=True,
+            timeout=TIMEOUT,
+        )
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_log_rotation(ops_test: OpsTest) -> None:
```

Comment:
nice test - I wonder if we could merge it with another one so we avoid a redeployment (I've included some quick tests on `test_replication.py` like testing for metrics endpoint and variable setting)